prediction_market_agent_tooling/markets/omen/omen_subgraph_handler.py

```diff
@@ -505,3 +525,26 @@ def get_answers(self, question_id: HexBytes) -> list[RealityAnswer]:
         result = self.sg.query_json(fields)
         items = self._parse_items_from_json(result)
         return [RealityAnswer.model_validate(i) for i in items]
+
+    def get_markets_from_all_user_positions(
+        self, user_positions: list[OmenUserPosition]
+    ) -> list[OmenMarket]:
+        condition_ids: list[HexBytes] = sum(
```

Comment:
There can be potentially many duplicates

```suggestion
        condition_ids: list[HexBytes] = list(set(sum(
```

Reply:
Great catch, yes for sure duplicates should be expected. Implemented.